modules/hdf5_handler/h5dmr.cc

```diff
@@ -1854,18 +1854,52 @@ hsize_t obtain_unlim_pure_dim_size_internal_value(hid_t dset_id, hid_t attr_id,
             throw InternalErr(__FILE__, __LINE__, msg);
         }
 
-        hssize_t num_ele_dim = H5Sget_simple_extent_npoints(did_space);
-        if (num_ele_dim < 0) {
+        // Check if this is a simple 
+        if (H5Sget_simple_extent_type(did_space) != H5S_SIMPLE) {
             H5Aclose(attr_id);
             H5Tclose(atype_id);
             H5Dclose(dset_id);
             H5Dclose(did_ref);
-            H5Sclose(did_space);
-            string msg = "Cannot obtain the number of elements for space of the attribute  " + reference_name;
+            string msg = "The dataspace must be a simple HDF5 dataspace for the variable  " + dname;
             throw InternalErr(__FILE__, __LINE__, msg);
         }
 
-        ret_value =(hsize_t)num_ele_dim;
+        int did_space_num_dims = H5Sget_simple_extent_ndims(did_space);
+        if (did_space_num_dims <0) {
+            H5Aclose(attr_id);
+            H5Tclose(atype_id);
+            H5Dclose(dset_id);
+            H5Dclose(did_ref);
+            string msg = "The number of dimensions must be > 0 for the variable  " + dname;
+            throw InternalErr(__FILE__, __LINE__, msg);
+        }
+        vector<hsize_t> did_dims;
```

Comment:
You can shorten the line count by initializing the vectors like 'vector<hsize_t> did_dims(did_space_num_dims);' which might be a bit faster, too.

Reply:
Good suggestion. Will do.